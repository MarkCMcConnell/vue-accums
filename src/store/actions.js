// Root Actions
const actions = {
    CHANGE_VIEW ({ commit }, payload) {
        commit('changeView', payload)
    },
}

export default {
    actions
}