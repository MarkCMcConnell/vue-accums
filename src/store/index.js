import Vue from 'vue'
import Vuex from 'vuex'
// Import mock data
import memberData from '../../data/SampleElig.js'
import claims from '../../data/SampleClaims.js'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    memberData: memberData,
    claims: claims,
    currentView: 'Home'
  },
  mutations: {
    CHANGE_VIEW (state, payload) {
      state.currentView = payload
    }
  },
  actions: {
    CHANGE_VIEW ({ commit }, payload) {
      commit('changeView', payload)
    }
  },
  getters: {

  }
})
