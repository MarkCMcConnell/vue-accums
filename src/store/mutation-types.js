// Modal types
export const HIDE_MODAL = 'HIDE_MODAL'

// Navbar types
export const CHANGE_VIEW = 'CHANGE_VIEW'
