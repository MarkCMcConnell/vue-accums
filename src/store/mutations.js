// Root Mutations
const mutations = {
    CHANGE_VIEW (state, payload) {
        state.currentView = payload
    },
}

export default {
    mutations
}