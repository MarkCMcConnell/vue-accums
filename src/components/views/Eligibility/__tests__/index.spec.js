/* eslint-disable */
import Vuex from 'vuex'
import {
    mount,
    createLocalVue
} from '@vue/test-utils'
import store from '@/store'

import Elig from '@/components/views/Eligibility/Index.vue'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('Eligibility.vue', () => {
    let mockStore
    let wrapper

    beforeEach(() => {
        mockStore = store
        wrapper = mount(Elig, {
            store: mockStore,
            localVue,
            stubs: {
                PersonalInfo: '<div class="stubbed" />',
                EmploymentInfo: '<div class="stubbed" />',
                ContactInfo: '<div class="stubbed" />',
                PlanInfo: '<div class="stubbed" />',
                DataTable: '<div class="stubbed" />'
            }
        })
    })

    it('renders the component', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    describe('formatAccumulators', () => {
        describe('the return value', () => {
            it('should return an array', () => {
                expect(Array.isArray(wrapper.vm.formatAccumulators)).toBeTruthy()
            })

            it('should have an object in the array', () => {
                expect.arrayContaining([{}])
            })
        })

        it('converts IndividualAccum to Individual', () => {
            expect.arrayContaining([
                expect.objectContaining({
                    IndividualAccum: 'Individual'
                })
            ])
        })

        it('converts to CoverageType to full word', () => {
            expect.arrayContaining([
                expect.objectContaining({
                    CoverageType: 'Medical'
                })
            ])
        })

        it('converts values to currency amounts', () => {
            expect.arrayContaining([
                expect.objectContaining({
                    CurrentAmount: '$12323'
                })
            ])
        })
    })
})