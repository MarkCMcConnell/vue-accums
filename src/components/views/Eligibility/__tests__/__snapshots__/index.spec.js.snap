// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eligibility.vue renders the component 1`] = `
<section>
  <div class="row">
    <div class="col-md-10 offset-md-1">
      <div class="stubbed"></div>
    </div>
  </div>
  <div class="row">
    <div class="col-md-10 offset-md-1">
      <div class="stubbed"></div>
    </div>
  </div>
  <div class="row">
    <div class="col-md-10 offset-md-1">
      <div class="stubbed"></div>
    </div>
  </div>
  <div class="row">
    <div class="col-md-10 offset-md-1">
      <div class="stubbed"></div>
    </div>
  </div>
  <div class="row">
    <div class="col-md-10 offset-md-1">
      <h3>Coverages</h3>
      <div class="stubbed"></div>
    </div>
  </div>
  <div class="row">
    <div class="col-md-10 offset-md-1">
      <section>
        <h3>My Balances</h3>
        <div class="stubbed"></div>
      </section>
    </div>
  </div>
  <div class="row">
    <div class="col-md-10 offset-md-1">
      <h3>Gaps in Care</h3>
      <div class="stubbed"></div>
    </div>
  </div>
</section>
`;
